diff --git a/foreman/api/client.go b/foreman/api/client.go
--- a/foreman/api/client.go
+++ b/foreman/api/client.go
@@ -488,12 +488,13 @@ func (client *Client) WrapJSONWithTaxonomy(name interface{}, item interface{}) (
 		return nil, err
 	}
 
-	// Workaround for Foreman versions < 1.21 in case no default location/organization was defined for resources
-	//if client.clientConfig.LocationID >= 0 && client.clientConfig.OrganizationID >= 0 {
-	//	wrapped["location_id"] = client.clientConfig.LocationID
-	//	wrapped["organization_id"] = client.clientConfig.OrganizationID
-	//	log.Debugf("client.go#WrapJSONWithTaxonomy: item %+v", wrapped)
-	//}
+	if client.clientConfig.LocationID > 0 {
+		wrapped["location_id"] = client.clientConfig.LocationID
+	}
+	if client.clientConfig.OrganizationID > 0 {
+		wrapped["organization_id"] = client.clientConfig.OrganizationID
+	}
+	log.Debugf("client.go#WrapJSONWithTaxonomy: item %+v", wrapped)
 
 	return json.Marshal(wrapped)
 }
